src/AmbientSounds.Uwp/Services/SoundDataProvider.cs

```diff
@@ -210,14 +218,19 @@ private async Task WriteCacheAsync()
 
         private async Task<List<Sound>> GetPackagedSoundsAsync()
         {
-            if (_packagedSoundCache == null)
+            if (_packagedSoundCache is null)
             {
                 StorageFolder appInstalledFolder = Windows.ApplicationModel.Package.Current.InstalledLocation;
                 StorageFolder assets = await appInstalledFolder.GetFolderAsync("Assets");
                 StorageFile dataFile = await assets.GetFileAsync(DataFileName);
                 using Stream dataStream = await dataFile.OpenStreamForReadAsync();
                 var sounds = await JsonSerializer.DeserializeAsync<List<Sound>>(dataStream);
 
+                // TODO: throw or just ignore?
+                // A null sound means the serialization failed, but since this was a local asset it likely
+                // is something we want to detect as soon as possible with a hard crash, before a release?
+                Guard.IsNotNull(sounds, nameof(sounds));
```

Comment:
iirc, if the deserialization fails, it throws an error anyway. I don't think this is necessary, so we can remove

Reply:
Huh, that's interesting [`DeserializeAsync<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserializeasync) is still marked as returning a `ValueTask<TValue?>`, which would suggest the returned deserialized value might in fact be `null` in some cases. Not entirely sure when that would be though, since there's also a documented list of exceptions it should throw as you mentioned if the desserialization fails, yeah. Will ask for more information as I'm curious (or will just investigate in the source to understand this), but anyway with respect to your review comment I'll just remove this `Guard` call then, sure! 🙂